include/ginkgo/core/base/math.hpp

```diff
@@ -627,7 +657,15 @@ GKO_ATTRIBUTES GKO_INLINE constexpr T imag(const T &)
  * @return  conjugate of the object (by default, the object itself)
  */
 template <typename T>
-GKO_ATTRIBUTES GKO_INLINE T conj(const T &x)
+GKO_ATTRIBUTES GKO_INLINE xstd::enable_if_t<is_complex<T>(), remove_complex<T>>
```

Comment:
Conjugate is still complex:
```suggestion
GKO_ATTRIBUTES GKO_INLINE xstd::enable_if_t<is_complex<T>(), T>
```